docs/source/reference/api-server/api-server-upgrade.rst

```diff
@@ -0,0 +1,217 @@
+
+Step 5: Verify the upgrade
+--------------------------
+
+Verify the API server is able to serve requests and the version is consistent with the version you upgraded to:
+
+.. code-block:: console
+
+    $ sky api info
+    Using SkyPilot API server: <URL>
+    ├── Status: healthy, commit: <COMMIT>, version: <VERSION>
+    └── User: <USER> (<USER_ID>)
+
+If possible, you can also trigger your pipelines that depend on the API server to verify there is no compatibility issue after the upgrade.
+
+.. _sky-api-server-vm-upgrade:
+
+Upgrade the API server deployed on VM
+-------------------------------------
+
+.. note::
+
+    VM deployment does not offer graceful upgrading support. We recommend to use Helm deployment :ref:`sky-api-server-deploy` in production environments.
+
+Suppose the cluster name of the API server is ``api-server`` (which is used in the :ref:`sky-api-server-cloud-deploy` guide), you can upgrade the API server with the following steps:
+
+1. Switch to the original API server that is used to launch the API server VM instance. It is usually the local API server that was launched automatically when you ran ``sky launch`` to launch the API server VM instance in the :ref:`sky-api-server-cloud-deploy` guide:
+
+.. code-block:: bash
+
+    # Replace http://localhost:46580 with the real API server endpoint if you were not using the local API server to launch the API server VM instance.
+    sky api login -e http://localhost:46580
+
+2. Check the API server VM instance is ``UP``:
+
+.. code-block:: console
+
+    $ sky status --no-show-managed-jobs --no-show-services
+    Clusters
+    NAME        LAUNCHED     RESOURCES                                                                  STATUS  AUTOSTOP  COMMAND
+    api-server  41 mins ago  1x AWS(c6i.2xlarge, image_id={'us-east-1': 'docker:berkeleyskypilot/sk...  UP      -         sky exec api-server pip i...
+
+3. Upgrade the SkyPilot pip package on the VM and restart the API server:
+
+.. note::
+
+    Upgrade and restart the API server will interrupt all running requests and may cause compatibility issue that requires upgrading the clients to resolve.
+
+.. code-block:: bash
+
+    sky exec api-server "pip install -U skypilot-nightly[all] && sky api stop && sky api start --deploy"
+    # Alternatively, you can also upgrade to a specific version with:
+    sky exec api-server "pip install -U skypilot-nightly[all]==${VERSION} && sky api stop && sky api start --deploy"
+
+4. Switch back to the remote API server:
+
+.. code-block:: bash
+
+    sky api login -e http://$(sky status --endpoint 46580 api-server)
+
+5. Verify the API server is running and the version is consistent with the version you upgraded to:
+
+.. code-block:: console
+
+    $ sky api info
+    Using SkyPilot API server: <URL>
+    ├── Status: healthy, commit: <COMMIT>, version: <VERSION>
+    └── User: <USER> (<USER_ID>)
```

Comment:
Let's just use an actual commit and version here. For the user we can show some mocked one, e.g. `alice (abcd1234)`

Reply:
Good call!